identity/resource_user.go

```diff
@@ -0,0 +1,82 @@
+package identity
+
+import (
+	"context"
+	"log"
+
+	"github.com/databrickslabs/databricks-terraform/common"
+	"github.com/databrickslabs/databricks-terraform/internal"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Bind entity to data
+func Bind(entity interface{}, err error) func(d *schema.ResourceData, s map[string]*schema.Schema) diag.Diagnostics {
+	return func(d *schema.ResourceData, s map[string]*schema.Schema) diag.Diagnostics {
+		if e, ok := err.(common.APIError); ok && e.IsMissing() {
+			log.Printf("missing resource due to error: %v\n", e)
+			d.SetId("")
+			return nil
+		}
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = internal.StructToData(entity, s, d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return nil
+	}
+}
```

Comment:
Hmm seems a bit obscure to make this a public method and only use it in the readContext. The method definition is also quite confusing to read in the readContext.

maybe something like:
```go
entity = NewUsersAPI(m).ReadR(d.Id())
return Bind(entity)(d, userSchema)
```

Reply:
removed this method